### PR TITLE
please.sh: drop the obsolete `sync` command

### PR DESCRIPTION
**Pull request by dscho**
Originally designed to keep clones of `git-sdk-32` and `git-sdk-64` in a private VM up to date, over the course of the years this command became more complex and also the driver of a CI job, which was then split into two, ported from an Azure Pipeline to a GitHub workflow, and eventually copy/edited into stand-alone GitHub workflows implemented mainly in PowerShell, with minimal dependencies, in `git-sdk-32`, `git-sdk-64` and `git-sdk-arm64`.

The subcommand itself is now superseded by these workflows.

This closes https://github.com/git-for-windows/git/issues/4234

### CODE REVIEW

**Review by derrickstolee — APPROVED** [review]

**Review by dennisameling — COMMENTED** [review]
Thanks! 🚀 

Just curious: as GfW is also [maintaining](https://github.com/git-for-windows/git/releases/tag/v2.35.6.windows.1) the 2.35.x version, will our recent changes to `please.sh` and other scripts break anything? Or will that use our new setup/scripts as well?

**Comment by dscho** [comment]
> Just curious: as GfW is also [maintaining](https://github.com/git-for-windows/git/releases/tag/v2.35.6.windows.1) the 2.35.x version, will our recent changes to `please.sh` and other scripts break anything? Or will that use our new setup/scripts as well?

The MinGit backports are done in a minimal fashion. That is, they use what I call "time-traveling" branches of `git-sdk-32`, `git-sdk-64`, `MINGW-packages` and `build-extra` that branch off of the revision used to build the previous MinGit version of that release train.

Besides, the `sync` subcommand is not even used in any Azure Pipeline or GitHub workflow. The last user were the `sync` workflows that we patched.

In other words: MinGit backports aren't broken by this PR ;-)
